16_wordle/main.go

```diff
@@ -99,3 +99,46 @@ func checkChar(chosenWordSlice []string, inputSlice []string) []string {
 	}
 	return containedSlice
 }
+
+func checkCharPrecise(chosenWordSlice []string, inputSlice []string) []string {
+	var containedSlice []string
+	// 1 = contained but incorrect loc
+	// 2 = contained in correc loc
+	// 0 not contained
+
+	// correct word check
+	if slices.Equal(chosenWordSlice, inputSlice) {
+		return []string{"Correct word!"}
+	}
+
+	for iInput, inputValue := range inputSlice {
+		// if inputValue == chosenValue {
+		// 	// valueInt, _ := strconv.Atoi(inputValue)
+		// 	containedSlice = append(containedSlice, 2)
+		// }
+		for iChosen, chosenValue := range chosenWordSlice {
+			if iInput == iChosen {
+				if inputValue == chosenValue {
+					// valueInt, _ := strconv.Atoi(inputValue)
+					containedSlice = append(containedSlice, inputValue+" inplace")
```

Comment:
Storing the `inputValue` here is useless, because you can get it from the index of `containedSlice`.

Another reason why storing `[]string` is making this more complex than you need it to.